### PR TITLE
feat(tree): adds a `changed` event on `TreeBranchEvents`

### PR DESCRIPTION
**Pull request by jenn-le**
adds a new `changed` event to `TreeBranchEvents` that fires for both local and remote changes

### CODE REVIEW

File: packages/dds/tree/src/shared-tree/schematizingTreeView.ts [hunk 1]
```diff
@@ -133,8 +134,8 @@ export class SchematizingSimpleTreeView<
 		this.update();
 
 		this.unregisterCallbacks.add(
-			this.checkout.events.on("commitApplied", (data, getRevertible) =>
-				this.events.emit("commitApplied", data, getRevertible),
+			this.checkout.events.on("changed", (data, getRevertible) =>
+				this.events.emit("changed", data, getRevertible),
```
Comment (noencke) [inline packages/dds/tree/src/shared-tree/schematizingTreeView.ts:138]: we still need to fire "commitApplied" as well, right?

File: packages/dds/tree/src/simple-tree/api/tree.ts [hunk 2]
```diff
@@ -619,6 +621,16 @@ export interface TreeBranchEvents {
 	 */
 	schemaChanged(): void;
 
+	/**
+	 * Fired when a change is made to the branch. Includes data about the change that is made which listeners
+	 * can use to filter on changes they care about e.g. local vs remote changes.
```
Comment (noencke) [inline packages/dds/tree/src/simple-tree/api/tree.ts:626]:
```suggestion
	 * can use to filter on changes they care about (e.g. local vs. remote changes).
```

File: packages/dds/tree/src/simple-tree/api/tree.ts [hunk 3]
```diff
@@ -619,6 +621,16 @@ export interface TreeBranchEvents {
...
+	 *
+	 * @param data - information about the change
+	 * @param getRevertible - a function provided that allows users to get a revertible for the change. If not provided,
```
Comment (noencke) [inline packages/dds/tree/src/simple-tree/api/tree.ts:629]:
```suggestion
	 * @param getRevertible - a function that allows users to get a revertible for the change. If not provided,
```

File: packages/dds/tree/src/simple-tree/api/tree.ts [hunk 4]
```diff
@@ -631,6 +643,9 @@ export interface TreeBranchEvents {
 	 * @param data - information about the commit that was applied
 	 * @param getRevertible - a function provided that allows users to get a revertible for the commit that was applied. If not provided,
 	 * this commit is not revertible.
+	 *
+	 * @deprecated use the "changed" event instead which behaves the same way but also gets fired for more scenarios, see its doc comment
```
Comment (noencke) [inline packages/dds/tree/src/simple-tree/api/tree.ts:647]: You can probably link here with {@link TreeBranchEvents.changed | "changed"}

File: packages/dds/tree/src/shared-tree/schematizingTreeView.ts (hunk 1 above)
Comment (noencke) [reply packages/dds/tree/src/shared-tree/schematizingTreeView.ts:138]: And retain test coverage until it's actually removed

File: packages/dds/tree/src/core/rebase/types.ts [hunk 5]
```diff
@@ -167,6 +167,10 @@ export enum CommitKind {
 	Undo,
 	/** A commit that is the result of a redo. */
 	Redo,
+	/**
+	 * A commit for which the context in which it was created cannot be determined.
+	 */
+	Unknown,
```
Comment (noencke) [inline packages/dds/tree/src/core/rebase/types.ts:173]: As discussed offline, remove in favor of default

File: packages/dds/tree/src/core/rebase/types.ts [hunk 6]
```diff
@@ -161,11 +161,11 @@ export interface GraphCommit<TChange> {
  * @public
  */
 export enum CommitKind {
-	/** A commit corresponding to a change that is not the result of an undo/redo. */
+	/** A commit corresponding to a change that is not the result of an undo/redo from this client. */
 	Default,
-	/** A commit that is the result of an undo. */
+	/** A commit that is the result of an undo from this client. */
 	Undo,
```
Comment (noencke) [inline packages/dds/tree/src/core/rebase/types.ts:167]: Update these to say "a commit that is the result of an undo/redo from this client" or similar

File: packages/dds/tree/src/shared-tree/schematizingTreeView.ts (hunk 1 above)
Comment (jenn-le) [reply packages/dds/tree/src/shared-tree/schematizingTreeView.ts:138]: oh right, yes

File: .changeset/stinky-cars-fart.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
+---
+"@fluidframework/tree": minor
+---
+---
+"section": tree
+---
+
+Added a new `changed` event to the (currently alpha) `TreeBranchEvents` that is meant to replace the `commitApplied` event on `TreeViewEvents`.
```
Comment (jzaffiro) [inline .changeset/stinky-cars-fart.md:8]:
```suggestion
Adds a new `changed` event to the (currently alpha) `TreeBranchEvents` that replaces the `commitApplied` event on `TreeViewEvents`.
```

File: .changeset/stinky-cars-fart.md [hunk 8]
```diff
@@ -0,0 +1,11 @@
+---
+"@fluidframework/tree": minor
+---
+---
+"section": tree
+---
+
```
Comment (jzaffiro) [inline .changeset/stinky-cars-fart.md:7]: Can you add a concise description of the change? Examples can be found in #22965

**Review by jzaffiro — APPROVED** [review]
Docs look good!

File: packages/dds/tree/src/shared-tree/schematizingTreeView.ts (hunk 1 above)
Comment (jenn-le) [reply packages/dds/tree/src/shared-tree/schematizingTreeView.ts]: so there weren't any tests specifically testing this event, just ones related to undo, so I added an undo test for just the commitApplied event

File: packages/dds/tree/api-report/tree.public.api.md [hunk 9]
```diff
@@ -580,6 +580,7 @@ export class TreeViewConfiguration<const TSchema extends ImplicitFieldSchema = I
 
 // @public @sealed
 export interface TreeViewEvents {
+    // @deprecated
```
Comment (noencke) [inline packages/dds/tree/api-report/tree.public.api.md:583]: Let's hold off on deprecating this for now. I'm not sure if we should deprecate something in the public API when the only alternative is an alpha API. Some users might not want to participate in alpha APIs, and we would be forcing them to choose between two evils. 

If we change our mind, we can easily deprecate later, no harm done, and we will at the very latest deprecate when the other event becomes public.

**Review by noencke — APPROVED** [review]

**Pull request merged**